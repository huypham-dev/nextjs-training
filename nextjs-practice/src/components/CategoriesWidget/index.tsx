import { memo } from "react";

import Link from "next/link";
import { CategoryType } from "types";
import isEqual from "react-fast-compare";

type Props = {
  data: CategoryType[];
};

const CategoriesWidget = ({ data = [] }: Props) => (
  <div className="w-full md:w-64">
    <aside className="rounded shadow overflow-hidden">
      <h3 className="text-sm bg-gray-100 text-gray-700 py-3 px-4 border-b">
        Categories
      </h3>

      <div className="p-4">
        <ul className="list-reset leading-normal">
          {data.map((item) => (
            <li key={item.id}>
              <Link href={`/blog/${item.name}`}>
                <a className="text-gray-darkest text-sm hover:underline">
                  {item.name.toUpperCase()}
                </a>
              </Link>
            </li>
          ))}
        </ul>
      </div>
    </aside>
  </div>
);

export default memo(CategoriesWidget, isEqual);
