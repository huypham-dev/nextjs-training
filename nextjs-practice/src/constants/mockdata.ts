export const CATEGORIES = [
  {
    id: "1",
    name: "mobile",
    image:
      "https://images.prismic.io/nextjsexample/7853f863-6a12-4312-a6da-f930437f6d62_mobile.png",
  },
  {
    id: "2",
    name: "internet",
    image:
      "https://images.prismic.io/nextjsexample/cb71ce70-bb03-442b-99e8-a318255b81f3_3579187.png",
  },
];

export const ARTICLES = [
  {
    id: "1",
    createdAt: "2021-03-01T12:59:16.855Z",
    title: "Lorem ipsum dolor sit amet, consectetur adipisicing elit",
    author: "author 1",
    image:
      "https://genk.mediacdn.vn/139269124445442048/2021/3/2/photo-1-16146533712881100681556.jpg",
    category: "internet",
    content:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.1",
    slug: "slug 1",
    view: 90,
  },
  {
    id: "2",
    createdAt: "2021-03-01T12:59:16.855Z",
    title: "Lorem ipsum dolor sit amet, consectetur adipisicing elit",
    author: "author 1",
    image:
      "https://genk.mediacdn.vn/139269124445442048/2021/3/2/photo-1-16146533712881100681556.jpg",
    category: "internet",
    content:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.1",
    slug: "slug 1",
    view: 90,
  },
];
