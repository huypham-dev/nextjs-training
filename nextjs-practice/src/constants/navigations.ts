export const navigations = [
  { label: "Home", path: "/" },
  { label: "Blog", path: "/blog" },
  { label: "Contact", path: "/contact" },
];
