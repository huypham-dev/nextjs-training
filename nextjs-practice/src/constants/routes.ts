export const ROUTES = {
  home: {
    path: "/",
  },
  blog: {
    path: "/blog",
    withId: (id: string) => `/blog/${id}`,
  },
  post: {
    path: "/post",
    withId: (id: string) => `/post/${id}`,
  },
  contact: {
    path: "/",
  },
};
