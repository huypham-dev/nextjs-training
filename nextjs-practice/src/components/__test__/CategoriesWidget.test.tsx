import { mount, shallow } from "enzyme";
import CategoriesWidget from "../CategoriesWidget";
import { mockCategories } from "constants/mockdata";

describe("CategoriesWidget", () => {
  it("Snapshot", () => {
    const component = shallow(<CategoriesWidget data={[]} />);
    expect(component).toMatchSnapshot();
  });

  it("Render component correctly with props", () => {
    const component = mount(<CategoriesWidget data={mockCategories} />);
    const links = component.find("a");

    expect(component.find("h3").text()).toBe("Categories");
    expect(component.find("a").length).toBe(2);
    expect(links.at(0).text()).toBe("MOBILE");
    expect(links.at(1).text()).toBe("INTERNET");
  });
});
