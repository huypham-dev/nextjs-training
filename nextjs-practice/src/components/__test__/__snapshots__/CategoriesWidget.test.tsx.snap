// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesWidget Snapshot 1`] = `
<div
  className="w-full md:w-64"
>
  <aside
    className="rounded shadow overflow-hidden"
  >
    <h3
      className="text-sm bg-gray-100 text-gray-700 py-3 px-4 border-b"
    >
      Categories
    </h3>
    <div
      className="p-4"
    >
      <ul
        className="list-reset leading-normal"
      />
    </div>
  </aside>
</div>
`;
